crates/wasi-common/src/old/snapshot_0/sys/windows/hostcalls_impl/fs_helpers.rs

```diff
@@ -49,31 +50,30 @@ pub(crate) fn openat(dirfd: &File, path: &str) -> Result<File> {
     use std::fs::OpenOptions;
     use std::os::windows::fs::OpenOptionsExt;
     use winx::file::Flags;
-    use winx::winerror::WinError;
 
     let path = concatenate(dirfd, Path::new(path))?;
     OpenOptions::new()
         .read(true)
         .custom_flags(Flags::FILE_FLAG_BACKUP_SEMANTICS.bits())
         .open(&path)
-        .map_err(|e| match e.raw_os_error() {
-            Some(e) => {
-                log::debug!("openat error={:?}", e);
-                match WinError::from_u32(e as u32) {
-                    WinError::ERROR_INVALID_NAME => Error::ENOTDIR,
-                    e => e.into(),
+        .map_err(|err| match err.raw_os_error() {
```

Comment:
One thing I've found helpful personally is to consistently use the same error handling pattern, e.g. using `match` everywhere or using `map_err` everywhere. (I realize this is existing code so don't put too much weight on this, just wanted to leave a passing comment about this)

Basically I think it'd be better to use the same idioms here (not `map_err`) to handle errors as elsewhere.

Reply:
Oh oops, the reply to the above should have been written down here actually.